File: app/src/main/java/com/hippo/util/DataUtils.java
package com.hippo.util;

import android.os.Parcel;
import android.os.Parcelable;

import com.hippo.ehviewer.dao.DownloadInfo;

public class DataUtils {
    public static <T> T copy(Parcelable input){
        Parcel parcel = null;
        try{
            parcel = Parcel.obtain();
            parcel.writeParcelable(input,0);
            parcel.setDataPosition(0);
            return parcel.readParcelable(input.getClass().getClassLoader());
        }finally {
            parcel.recycle();
        }
    }

}

File: app/src/main/java/com/hippo/util/DataUtils.kt
package com.hippo.util

import android.os.Parcel
import android.os.Parcelable

object DataUtils {
    @JvmStatic
    fun <T> copy(input: Parcelable): T? {
        var parcel: Parcel? = null
        try {
            parcel = Parcel.obtain()
            parcel.writeParcelable(input, 0)
            parcel.setDataPosition(0)
            return parcel.readParcelable(input.javaClass.getClassLoader())
        } finally {
            parcel!!.recycle()
        }
    }
}
